Panicked on linux

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
<!--If possible, provide a recipe for reproducing the error.-->

```
./bin/pairmesh -k <key>
```


2. What did you expect to see?

Normal running

3. What did you see instead?4. 

```
2022-04-04T05:40:29.736+0800    WARN    Client connection transporter terminated        {"addr": "139.198.29.15:2328"}
panic: send on closed channel

goroutine 38 [running]:
github.com/pairmesh/pairmesh/internal/relay.(*Client).Send(0xc000280300, 0x2, {0x961280, 0xc000280440})
        /home/lonng/devel/pairmesh/pairmesh/internal/relay/client.go:63 +0xb8
github.com/pairmesh/pairmesh/node/mesh.(*Manager).probePeers(0xc00059c280)
        /home/lonng/devel/pairmesh/pairmesh/node/mesh/manager.go:273 +0x37c
github.com/pairmesh/pairmesh/node/mesh.(*Manager).Tick(...)
        /home/lonng/devel/pairmesh/pairmesh/node/mesh/manager.go:167
github.com/pairmesh/pairmesh/node/driver.(*NodeDriver).pullPeerGraph(0xc00051c120, {0x96bef8, 0xc0002de540})
        /home/lonng/devel/pairmesh/pairmesh/node/driver/pull_peers.go:84 +0x445
created by github.com/pairmesh/pairmesh/node/driver.(*NodeDriver).Drive
        /home/lonng/devel/pairmesh/pairmesh/node/driver/driver.go:218 +0x191
```

5. What version of PairMesh are you using (`PairMesh --version`)?

--- a/internal/relay/client.go
+++ b/internal/relay/client.go
@@ -54,17 +54,30 @@ func (c *Client) Handler() ClientHandler {
 	return c.handler
 }
 
-func (c *Client) Send(typ message.PacketType, msg proto.Message) error {
+func (c *Client) Send(typ message.PacketType, msg proto.Message) (err error) {
 	if c.closed.Load() {
 		return errors.New("cannot send message to closed client")
 	}
+
+	// Prevents write data to a closed channel
+	defer func() {
+		if e := recover(); e != nil {
+			e1, ok := e.(error)
+			if !ok {
+				err = fmt.Errorf("send data panicked: %v", e)
+			} else {
+				err = e1
+			}
+		}
+	}()
 
 	select {
 	case c.WriteQueue() <- Packet{Type: typ, Message: msg}:
 		return nil
 	default:
-		return fmt.Errorf("send buffer exceeded: %s:%d", c.RelayServer().Host, c.RelayServer().Port)
+		err = fmt.Errorf("send buffer exceeded: %s:%d", c.RelayServer().Host, c.RelayServer().Port)
 	}
+	return
 }
 
 func (c *Client) Serve(ctx context.Context) {

--- a/internal/relay/session.go
+++ b/internal/relay/session.go
@@ -134,17 +134,30 @@ func (s *Session) LifetimeHook() SessionLifetimeHook {
 }
 
 // Send sends message to the pairmesh client-side via the client session.
-func (s *Session) Send(typ message.PacketType, msg proto.Message) error {
+func (s *Session) Send(typ message.PacketType, msg proto.Message) (err error) {
 	if s.closed.Load() {
 		return errors.New("cannot send message to closed session")
 	}
+
+	// Prevents write data to a closed channel
+	defer func() {
+		if e := recover(); e != nil {
+			e1, ok := e.(error)
+			if !ok {
+				err = fmt.Errorf("send data panicked: %v", e)
+			} else {
+				err = e1
+			}
+		}
+	}()
 
 	select {
 	case s.WriteQueue() <- Packet{Type: typ, Message: msg}:
 		return nil
 	default:
-		return fmt.Errorf("write buffer exceed: %d", s.peerID)
+		err = fmt.Errorf("write buffer exceed: %d", s.peerID)
 	}
+	return
 }
 
 func (s *Session) Serve(ctx context.Context, wg *sync.WaitGroup) {

